### PR TITLE
refactor(css/codegen): do not use `raw`

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Refactor codegen and do not use `raw`, also:
1. fix bug with printing `e\32px` dimension (we need to escape as in spec)
2. do not parser wrong hash token as color
3. fix especaping in `"` in `Str`
4. fix serialization in raw URL
5. support `element()` function (before we parse it as hash, now as id selector)

**BREAKING CHANGE:**

Yes, we don't use `raw` in codegen, except printing tokens, so if you want to inject something non standard, you should use `ListOfComponentValues` with `PreservedToken`

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_ast/src/base.rs [hunk 1]
```diff
@@ -190,6 +190,8 @@ pub enum ComponentValue {
     SupportsCondition(Box<SupportsCondition>),
     #[tag("Declaration")]
     Declaration(Box<Declaration>),
+    #[tag("IdSelector")]
+    IdSelector(Box<IdSelector>),
```
Comment (kdy1) [inline crates/swc_css_ast/src/base.rs:194]: Is this change really required?

**Comment by alexander-akait** [comment]
@kdy1 Rebased too

**Review by kdy1 — DISMISSED** [review]
swc-bump:
 - swc_css_ast --breaking

**Comment by alexander-akait** [comment]
@kdy1 Rebased too

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

**Comment by alexander-akait** [comment]
@kdy1 Fixed too :+1:

**Review by kdy1 — APPROVED** [review]

**Pull request merged**